fix(minifier): keep `Array` reference id when compressing `Array()`

sapphi-red opened the pull request:
When compressing `new Array(x)` to `Array(x)`, the reference id was removed from `Array`.

This PR fixes that.

Boshen commented:
### Merge activity

* **Aug 15, 11:24 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

sapphi-red committed:
fix(minifier): keep `Array` reference id when compressing `Array()` (#13113)

When compressing `new Array(x)` to `Array(x)`, the reference id was removed from `Array`.

This PR fixes that.
